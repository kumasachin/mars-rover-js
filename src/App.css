.App {
  text-align: left;
  margin: 20px;
  font-family: Arial, Helvetica, sans-serif;
}
html, body {
  height: 100%;
  margin: 0px;
}