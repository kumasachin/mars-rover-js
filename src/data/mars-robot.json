{
    "map":{
       "x":16,
       "y":16
    },
    "lostCell":{
       "x":[
          
       ],
       "y":[
          
       ]
    },
    "robots":[
       {
          "name":"R1",
          "color":"green",
          "currentPosition":"0 0 N",
          "instructions":"FFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
       },
       {
          "name":"R1_1",
          "color":"green",
          "currentPosition":"0 0 N",
          "instructions":"FFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
       },
      {
         "name":"R2",
         "color":"blue",
         "currentPosition":"1 12 E",
         "instructions":"FFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
      },
      {
         "name":"R2_1",
         "color":"blue",
         "currentPosition":"1 12 E",
         "instructions":"FFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
      },
      {
         "name":"R3",
         "color":"blue",
         "currentPosition":"4 7 S",
         "instructions":"FFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
      },
      {
         "name":"R3_1",
         "color":"blue",
         "currentPosition":"4 7 S",
         "instructions":"FFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
      },
      {
         "name":"R4",
         "color":"blue",
         "currentPosition":"12 3 W",
         "instructions":"FFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
      },
      {
         "name":"R4_1",
         "color":"blue",
         "currentPosition":"12 3 W",
         "instructions":"FFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
      },

      {
         "name":"R2_2",
         "color":"blue",
         "currentPosition":"1 12 E",
         "instructions":"FFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
      },
      {
         "name":"R2_1_2",
         "color":"blue",
         "currentPosition":"1 12 E",
         "instructions":"FFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
      },
      {
         "name":"R3_2",
         "color":"blue",
         "currentPosition":"4 7 S",
         "instructions":"FFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
      },
      {
         "name":"R3_1_2",
         "color":"blue",
         "currentPosition":"4 7 S",
         "instructions":"FFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
      },
      {
         "name":"R4_2",
         "color":"blue",
         "currentPosition":"12 3 W",
         "instructions":"FFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
      },
      {
         "name":"R4_1_2",
         "color":"blue",
         "currentPosition":"12 3 W",
         "instructions":"FFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
      }
    ]
 }