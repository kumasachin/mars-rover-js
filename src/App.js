import React, { useState } from 'react';
import { CSVReaderComponent, Planet } from './components/';
import { MarsContextProvider } from './context/marsContext';
import './App.css';

const App = ({ CSVTestData }) => {
  const [data, setData] = useState();
  const setCSVData = (response) => {
    setData(response);
  };

  return (
    <div className="App">
      <CSVReaderComponent setCSVData={setCSVData} CSVTestData={CSVTestData} />
      <MarsContextProvider data={data || CSVTestData}>
        <Planet />
      </MarsContextProvider>
    </div>
  );
};

export default App;
