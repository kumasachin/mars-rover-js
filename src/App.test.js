import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  cleanup,
} from '@testing-library/react';
import Enzyme, { mount } from 'enzyme';
import App from './App';
import data from './mocks/';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

describe('App snapshot match', () => {
  afterEach(cleanup);
  beforeEach(() => render(<App />));
  it('should take a snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment(<App />)).toMatchSnapshot();
  });
  it('CSV selector should visible without grid', () => {
    const container01 = mount(<App />);
    expect(container01.find('div.react-csv-input').length).toEqual(1);
    expect(container01.find('table.grid-container').length).toEqual(0);
  });
});

describe('App custom CSV file upload', () => {
  afterEach(cleanup);
  const container = mount(<App />);

  it('CSV selector should visible without grid', () => {
    expect(container.find('input[type="file"]').length).toEqual(1);
  });
});

describe('App CSV rendered grid', () => {
  const container = mount(<App CSVTestData={data} />);

  it('CSV selector should visible with grid', () => {
    expect(container.find('div.add-new-form').length).toEqual(1);
    expect(container.find('input[name="roboname"]').length).toEqual(1);
    expect(container.find('input[name="csvSubmit"]').length).toEqual(1);
  });
  it('CSV selector should visible with grid', () => {
    expect(container.find('table.grid-container').length).toEqual(1);
  });
  it('CSV selector robo should placed one by one', () => {
    expect(container.find('span.robot').length).toEqual(1);
  });
});
