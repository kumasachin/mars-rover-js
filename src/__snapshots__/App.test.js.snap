// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot match should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <p>
      Upload CSV with robots movements
    </p>
    <br />
    <div
      class="react-csv-input"
    >
      <input
        accept=".csv, text/csv"
        class="csv-input"
        id="react-csv-reader-input"
        name="react-csv-reader-input"
        type="file"
      />
    </div>
    <div
      class="planet-container"
    />
  </div>
</DocumentFragment>
`;
